drivers/storage/portworx/util/util.go

```diff
@@ -303,6 +305,39 @@ func GetPortworxVersion(cluster *corev1alpha1.StorageCluster) *version.Version {
 	return pxVersion
 }
 
+// GetStorkVersion returns the stork version based on the image provided.
+// We first look at spec.Image. If that is not present or invalid semvar, then we fallback to an
+// annotation portworx.io/stork-version; else we return int max as the version.
+func GetStorkVersion(cluster *corev1alpha1.StorageCluster) *version.Version {
+	defaultVersion, _ := version.NewVersion(strconv.FormatInt(math.MaxInt64, 10))
+	if cluster.Spec.Stork == nil && cluster.Annotations[AnnotationStorkVersion] == "" {
+		logrus.Warnf("could not find stork version from cluster spec")
+		return defaultVersion
+	}
+
+	parts := strings.Split(cluster.Spec.Stork.Image, ":")
```

Comment:
Starting operator 1.4, `cluster.Spec.Stork.Image` may not always be present. If it is present, it will be only to override the operator selected image. The operator selected image will be found in `cluster.Status.DesiredImages.Stork`.
Basically it should be - 
```
if cluster.Spec.Stork.Image == "" {
  // get version from cluster.Status.DesiredImages.Stork
  // if this is semver, return it
  // else get it from the annotation and return it
} else {
  // if this is semver, return it
  // else get it from the annotation and return it
}
```